### PR TITLE
Add orders feature and the tool tip form

### PR DESCRIPTION
**Pull request by Ammar-alnatsheh**
screen shot 2018-12-30 at 8 35 26 pm

screen shot 2018-12-30 at 8 35 55 pm

### CODE REVIEW

File: client/src/components/OrderList.jsx [hunk 1]
```diff
@@ -1,11 +1,51 @@
...
 
+function sanitizeOrderItems(props) {
+    if (props.order_items.slice(2,props.order_items.length-2).split('').length === 0) {
+        return [];
+    } else {
+        return props.order_items.slice(2,props.order_items.length-2).split(',');
+    }
+} 
+
+function sanitizeMenu(str) {
```
Comment (DymaxionMap) [inline client/src/components/OrderList.jsx]: What does this function do? It looks complicated. Is there a better name for it?

File: client/src/components/OrderList.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import OrderListEntity from './OrderListEntity.jsx';
+import style from '../style/style.css.jsx';
+
+function sanitizeOrderItems(props) {
```
Comment (DymaxionMap) [inline client/src/components/OrderList.jsx]: What does this function do?

File: client/src/components/OrderListEntity.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import style from '../style/style.css.jsx';
+
+class OrderListEntity extends React.Component {
+    constructor(props) {
+      super(props);
+  
+      this.state = {
+        visiable: false,
```
Comment (DymaxionMap) [inline client/src/components/OrderListEntity.jsx]:
```suggestion
        visible: false,
```

File: client/src/components/OrderListEntity.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            visiable: false
+        });
+    }
+    
+    render() {
+        return (
+        <div className="order-list-entity" style={style.orderListEntity} onMouseEnter={this.onMouseEnter.bind(this)} onMouseLeave={this.onMouseLeave.bind(this)}>
+            {this.props.item['name'] + ' + '}
+            {
+                this.state.visiable ?
```
Comment (DymaxionMap) [inline client/src/components/OrderListEntity.jsx]:
```suggestion
                this.state.visible ?
```

File: client/src/components/OrderListEntity.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                    <form className="tooltiptext" style={style.tooltiptext}>
+                        <div style={style.tooltiptextHeader}><span style={style.userName}>{this.props.item['name']}</span><span>${this.props.item['price']}</span></div>
+                        {
+                            this.props.item['content'].length === 0?
```
Comment (DymaxionMap) [inline client/src/components/OrderListEntity.jsx]: The question mark should be at the start of the second line below, and the div should be after the colon. See https://github.com/airbnb/javascript#comparison--nested-ternaries

File: client/src/components/OrderList.jsx (hunk 1 above)
Comment (Ammar-alnatsheh) [reply client/src/components/OrderList.jsx]: the problem with sql 5.7 is that when i get the JSON file from the server is that i cant get the the right array back, mysql the new version has the array feature implement in it but the old one it deal with arrays and object in a different way so i use this function and the sanitizeOrderItems to make sure i have the right formate to deal with it in react

File: client/src/components/OrderList.jsx (hunk 2 above)
Comment (Ammar-alnatsheh) [reply client/src/components/OrderList.jsx]: same concept as sanitizeMenu but the array formate is a little different

File: client/src/components/OrderListEntity.jsx (hunk 3 above)
Comment (Ammar-alnatsheh) [reply client/src/components/OrderListEntity.jsx]: spelling error :(

File: client/src/components/OrderListEntity.jsx (hunk 4 above)
Comment (Ammar-alnatsheh) [reply client/src/components/OrderListEntity.jsx]: i will fix the spelling error

File: client/src/components/OrderListEntity.jsx (hunk 5 above)
Comment (Ammar-alnatsheh) [reply client/src/components/OrderListEntity.jsx]: i will fix that

File: client/src/components/OrderList.jsx (hunk 1 above)
Comment (DymaxionMap) [reply client/src/components/OrderList.jsx]: OK. Maybe you can call it something like `parseMenu` instead? It takes a string and returns an object right? That would be clearer than `sanitizeMenu`. 

Also do you need 2 while loops below? The outer loop and inner loop both have the condition `menu.length > 0`, so I don't if there's something overlap between the two?

Comment (Ammar-alnatsheh) [reply client/src/components/OrderList.jsx]: the outer loop build the object keys and the inner loop is to build an array for all the content of each menu item

**Pull request merged**